--- a/src/common-config.h
+++ b/src/common-config.h
@@ -61,6 +61,18 @@
	int gid_min;
 } pam_cfg_st;

+typedef struct saml_cfg_st {
+	char *config;          /* SAML subconfig file path */
+	char *spmeta;          /* SP metadata file path */
+	char *spkey;           /* SP private key file path */
+	char *spcert;          /* SP certificate file path */
+	char *idpmeta;         /* IdP metadata file path */
+	char *idpname;         /* IdP name (runtime) */
+	char *idpcert;         /* IdP certificate file path */
+	char *idp_sso_dest_url; /* IdP SSO destination URL (runtime) */
+	char *acs_url;         /* Assertion Consumer Service URL (runtime) */
+} saml_cfg_st;
+
 #define CHECK_TRUE(str)                                                    \
	((str != NULL &&                                                   \
	  (strcasecmp(str, "true") == 0 || strcasecmp(str, "yes") == 0)) ? \
@@ -80,6 +92,8 @@
				const char *str);
 void *oidc_get_brackets_string(void *pool, struct perm_cfg_st *config,
			       const char *str);
+void *saml_get_brackets_string(void *pool, struct perm_cfg_st *config,
+			       const char *str);

 void parse_kkdcp_string(char *str, int *socktype, char **_port, char **_server,
			char **_path, char **_realm);