--- a/src/config.c
+++ b/src/config.c
@@ -37,6 +37,9 @@
 #include <auth/plain.h>
 #include <auth/gssapi.h>
 #include <auth/openidconnect.h>
+#ifdef HAVE_SAML
+#include <auth/saml.h>
+#endif
 #include <auth/common.h>
 #include <sec-mod-sup-config.h>
 #include <sec-mod-acct.h>
@@ -216,6 +219,10 @@
	{ NAME("oidc"), &oidc_auth_funcs, AUTH_TYPE_OIDC,
	  oidc_get_brackets_string },
 #endif
+#ifdef HAVE_SAML
+	{ NAME("saml"), &saml_auth_funcs, AUTH_TYPE_SAML,
+	  saml_get_brackets_string },
+#endif
 };

 static void check_for_duplicate_password_auth(struct perm_cfg_st *config,
@@ -1015,6 +1015,8 @@
 	} else if (strcmp(name, "default-domain") == 0) {
 		READ_STRING(config->default_domain);
+	} else if (strcmp(name, "hostname") == 0) {
+		READ_STRING(config->hostname);
 	} else if (strcmp(name, "crl") == 0) {
 		READ_STRING(config->crl);
 	} else if (strcmp(name, "cert-user-oid") == 0) {
