--- ocserv-1.4.2/meson.build	2026-04-16 21:27:59.000000000 +0000
+++ ocserv-1.4.2/meson.build.new	2026-06-04 14:38:30.427500003 +0000
@@ -75,6 +75,7 @@
 opt_seccomp = get_option('seccomp')
 opt_systemd = get_option('systemd')
 opt_oidc    = get_option('oidc-auth')
+opt_saml    = get_option('saml-auth')
 opt_utmp    = get_option('utmp')
 opt_wrap    = get_option('libwrap')

@@ -229,6 +230,20 @@
   oidc_enabled = curl_dep.found() and cjose_dep.found() and jansson_dep.found()
 endif

+# SAML2
+lasso_dep  = dependency('', required: false)
+glib_dep   = dependency('', required: false)
+gobject_dep = dependency('', required: false)
+apr_dep    = dependency('', required: false)
+saml_enabled = false
+if not opt_saml.disabled()
+  lasso_dep  = dependency('lasso',    version: '>= 2.5.1', required: opt_saml)
+  glib_dep   = dependency('glib-2.0', required: opt_saml)
+  gobject_dep = dependency('gobject-2.0', required: opt_saml)
+  apr_dep    = dependency('apr-1',    required: opt_saml)
+  saml_enabled = lasso_dep.found() and glib_dep.found() and gobject_dep.found() and apr_dep.found()
+endif
+
 # libwrap (TCP wrappers)
 wrap_dep = dependency('', required: false)
 if not opt_wrap.disabled()
@@ -362,6 +377,7 @@
 cdata.set('NETTLE_OLD_BASE64_API',            old_base64)
 cdata.set('PROC_FS_SUPPORTED',                proc_fs)
 cdata.set('SUPPORT_OIDC_AUTH',                oidc_enabled)
+cdata.set('HAVE_SAML',                   saml_enabled)
 cdata.set('TRY_SHA2_CRYPT',                   try_sha2_crypt)
 cdata.set('USE_SECCOMP_TRAP',                 get_option('seccomp-trap'))
