--- a/lasso/xml/tools.c
+++ b/lasso/xml/tools.c
@@ -311,7 +311,8 @@ xmlSecKeyPtr lasso_get_public_key_from_pem_file(const char *file)
             break;
         case LASSO_PEM_FILE_TYPE_PUB_KEY:
-            pub_key = xmlSecCryptoAppKeyLoad(file,
-                    xmlSecKeyDataFormatPem, NULL, NULL, NULL);
+            pub_key = xmlSecCryptoAppKeyLoadEx(file,
+                    xmlSecKeyDataTypePublic, xmlSecKeyDataFormatPem, NULL, NULL, NULL);
             break;
         case LASSO_PEM_FILE_TYPE_PRIVATE_KEY:
             pub_key = lasso_get_public_key_from_private_key_file(file);
@@ -379,7 +380,8 @@ static xmlSecKeyPtr
 lasso_get_public_key_from_private_key_file(const char *private_key_file)
 {
-    return xmlSecCryptoAppKeyLoad(private_key_file,
-            xmlSecKeyDataFormatPem, NULL, NULL, NULL);
+    return xmlSecCryptoAppKeyLoadEx(private_key_file,
+            xmlSecKeyDataTypePrivate, xmlSecKeyDataFormatPem, NULL, NULL, NULL);
 }

--- a/lasso/lasso.c
+++ b/lasso/lasso.c
@@ -276,6 +276,9 @@ int lasso_init(void)
     * xmlsec-crypto library.
     */
+#define XMLSEC_CRYPTO "openssl"
 #ifdef XMLSEC_CRYPTO_DYNAMIC_LOADING
     if (xmlSecCryptoDLLoadLibrary(BAD_CAST XMLSEC_CRYPTO) < 0) {
